src/main/java/com/limechain/utils/scale/ScaleUtils.java

```diff
@@ -69,5 +77,41 @@ public byte[] encode(byte[][] values) throws IOException {
                 return buffer.toByteArray();
             }
         }
+
+        public <T> byte[] encode(ScaleWriter<T> writer, T value) {
+            try (ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+                 ScaleCodecWriter scaleCodecWriter = new ScaleCodecWriter(buffer)) {
+                writer.write(scaleCodecWriter, value);
+                return buffer.toByteArray();
+            } catch (IOException e) {
+                throw new ScaleEncodingException("Unexpected exception while encoding.");
+            }
+        }
+
+        public <T> byte[] encodeAsList(ScaleWriter<T> writer, Collection<T> values) {
+            return encode(new CollectionWriter<>(writer), values);
+        }
+
+        public byte[] encodeAsListOfBytes(Collection<Byte> bytes) {
+            return encodeAsList(ScaleCodecWriter::directWrite, bytes);
+        }
+
+        public byte[] encodeAsListOfListsOfBytes(Collection<Collection<Byte>> values) {
+            return encodeAsList(new CollectionWriter<>(ScaleCodecWriter::directWrite), values);
+        }
+
+//        public byte[] encodeListOfListsOfBytes(List<List<Byte>> values) {
+//            return encodeAsList(new ListWriter<>(ScaleCodecWriter::directWrite), values);
+//        }
```

Comment:
Is this intentional?

Reply:
`ScaleUtils` remains to be cleaned up, this is not intentional.

Reply:
I've done all the refactoring necessary, but afterwards I thought about the breaking nature of some of my changes (main suspects: method renamings). Merging the refactored code in such a state would break the build on `dev`, so I'd propose we skip this part until we merge and apply the changes afterwards in a new PR to `dev`.

Reply:
That's what we'll do.